src/controllers/connections/ConnectionController.ts

```diff
@@ -170,4 +171,34 @@ export class ConnectionController extends Controller {
       throw error
     }
   }
+
+  /**
+   * @description Creates inbound out-of-band record from an implicit invitation, given as public DID the agent should be able to resolve.
+   * If any existing connections for the DID exists, delete and replace them with a single new one.
+   * @returns out-of-band record and connection record if one has been created.
+   */
+  @Post('/')
+  public async post(@Body() body: { did: string }) {
+    const { did } = body
+
+    const connectionRepository = this.agent.dependencyManager.resolve(ConnectionRepository)
+    const connections = await connectionRepository.findByQuery(this.agent.context, {
```

Comment:
I'd suggest we use the APIs as opposed to internals (Service, Repository, etc) as these are subject to semver, whereas internals can change at any time. I realise we use the internals elsewhere, even in this file, but in this particular case, the API exposes the functionality we need: https://github.com/openwallet-foundation/credo-ts/blob/main/packages/core/src/modules/connections/ConnectionsApi.ts#L553C16-L553C35